File: src/utils/update_municipality_names.py
#!/usr/bin/env python3
import csv

# Definiera mappningar för kommunnamn som behöver ändras
name_mappings = {
    'DalsEd': 'Dals-Ed',
    'Habo': 'Håbo',
    'MalungSalen': 'Malung-Sälen',
    'UpplandssBro': 'Upplands-Bro'
}

# Läs in original CSV-filen
rows = []
with open('data/education_data/education_by_municipality_2024.csv', 'r', encoding='utf-8') as file:
    reader = csv.reader(file, delimiter=';')
    header = next(reader)  # Spara header
    for row in reader:
        # Kolla om kommunnamnet behöver uppdateras
        if row[1] in name_mappings:
            row[1] = name_mappings[row[1]]  # Uppdatera kommunnamnet
            # Uppdatera även det normaliserade namnet
            row[-1] = row[1].lower().replace('å', 'a').replace('ä', 'a').replace('ö', 'o').replace('-', '')
        rows.append(row)

# Skriv till en ny fil
with open('data/education_data/education_by_municipality_2024.updated.csv', 'w', encoding='utf-8', newline='') as file:
    writer = csv.writer(file, delimiter=';')
    writer.writerow(header)
    writer.writerows(rows)

print("CSV-fil uppdaterad och sparad som 'education_by_municipality_2024.updated.csv'")
print("Kontrollera den nya filen innan du ersätter originalet.") 

File: src/utils/update-municipality-names.js
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

// Definiera mappningar för kommunnamn som behöver ändras
const nameMappings = {
    'DalsEd': 'Dals-Ed',
    'Habo': 'Håbo',
    'MalungSalen': 'Malung-Sälen',
    'UpplandssBro': 'Upplands-Bro'
};

// Sökvägar till filer
const inputPath = path.join(__dirname, '../../data/education_data/education_by_municipality_2024.csv');
const outputPath = path.join(__dirname, '../../data/education_data/education_by_municipality_2024.updated.csv');

// Läs in original CSV-filen
const fileContent = fs.readFileSync(inputPath, 'utf8');
const rows = fileContent.split('\n');

// Uppdatera kommunnamnen
const updatedRows = rows.map((row, index) => {
    if (index === 0) return row; // Behåll header-raden oförändrad
    
    const columns = row.split(';');
    if (columns.length < 2) return row; // Skippa tomma rader
    
    const kommunNamn = columns[1];
    if (nameMappings[kommunNamn]) {
        // Uppdatera kommunnamnet
        columns[1] = nameMappings[kommunNamn];
        // Uppdatera det normaliserade namnet (sista kolumnen)
        columns[columns.length - 1] = columns[1]
            .toLowerCase()
            .replace(/å/g, 'a')
            .replace(/ä/g, 'a')
            .replace(/ö/g, 'o')
            .replace(/[^a-z]/g, '');
    }
    
    return columns.join(';');
});

// Skriv till en ny fil
fs.writeFileSync(outputPath, updatedRows.join('\n'), 'utf8');

console.log('CSV-fil uppdaterad och sparad som "education_by_municipality_2024.updated.csv"');
console.log('Kontrollera den nya filen innan du ersätter originalet.'); 